api/python/notebooks/experimental/pytorch.ipynb

```diff
@@ -0,0 +1,675 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f874fb88",
+   "metadata": {},
+   "source": [
+    "## Open the Census\n",
+    "\n",
+    "First, obtain a handle to the Census data, in the usual manner:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "c3dd549f",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "The \"stable\" release is currently 2023-05-15. Specify 'census_version=\"2023-05-15\"' in future calls to open_soma() to ensure data consistency.\n"
+     ]
+    }
+   ],
+   "source": [
+    "import cellxgene_census\n",
+    "\n",
+    "census = cellxgene_census.open_soma()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "580b29f2",
+   "metadata": {},
+   "source": [
+    "## Create a DataLoader\n",
+    "\n",
+    "To train a model in PyTorch using this `census` data object, first instantiate an `ExperimentDataPipe` as follows:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "id": "54896e6f",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import cellxgene_census.experimental.ml as census_ml\n",
+    "import tiledbsoma as soma\n",
+    "\n",
+    "experiment_datapipe = census_ml.ExperimentDataPipe(\n",
+    "    census[\"census_data\"][\"homo_sapiens\"],\n",
+    "    measurement_name=\"RNA\",\n",
+    "    X_name=\"raw\",\n",
+    "    obs_query=soma.AxisQuery(value_filter=\"tissue_general == 'tongue' and is_primary_data == True\"),\n",
+    "    obs_column_names=[\"cell_type\"],\n",
+    "    batch_size=16,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6c7c17c3",
+   "metadata": {},
+   "source": [
+    "This class provides an implementation of PyTorch's `DataPipe` interface, which defines a common mechanism for wrapping and accessing training data from any underlying source. The `ExperimentDataPipe` class encapsulates the details of querying and retrieving Census data from a single SOMA `Experiment` and returning it to the caller as PyTorch Tensors. Most importantly, it retrieves the data lazily from the Census in batches, avoiding having to load the entire training dataset into memory at once.\n",
+    "\n",
+    "The constructor only requires a single parameter, `experiment`, which is a `soma.Experiment` containing the data of the organism to be used for training.\n",
+    "\n",
+    "To retrieve a subset of the Experiment's data, along either the `obs` or `var` axes, you may specify query filters via the `obs_query` and `var_query` parameters, which are both `soma.AxisQuery` objects.\n",
+    "\n",
+    "The values for the prediction label(s) that you intend to use for training are specified via the `obs_column_names` array. \n",
+    "\n",
+    "Finally, the `batch_size` allows you to specify the number of cells to return in each training iteration step. You may exclude this parameter if you want single rows (`batch_size=1`).\n",
```

Comment:
does batch size control matrix cells, or matrix rows.  The use of the term `cells` is confusing

Reply:
updated to say `number of obs rows (cells)`